ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js

```diff
@@ -40,38 +40,68 @@ export default function App(props) {
     const [networkError, setNetworkError] = useState(false);
 
     const getDefaultUploadAction = file => {
-        // disabled until we figure out a nicer algorithm
-        // const fuzzySearchResults = new Fuse(
-        //     updateResourcesOptions,
-        //     {
-        //         // https://bit.ly/3BTLnWd
-        //         includeScore: true,
-        //         findAllMatches: false,
-        //         keys: ['fileName'],
-        //         threshold: 0.3
-        //     }
-        // ).search(file.name);
-        // let defaultUploadAction = extraResource;
-        // if (fuzzySearchResults.length === 1) {
-        //     defaultUploadAction = fuzzySearchResults[0].item;
-        // } else if (fuzzySearchResults.length > 1) {
-        //     const firstResult = fuzzySearchResults[0];
-        //     const secondResult = fuzzySearchResults[1];
-        //     const scroreDiff = firstResult.score - secondResult.score;
-        //     if (scroreDiff > 0.05) {
-        //         defaultUploadAction = fuzzySearchResults[0].item;
-        //     }
-        // }
-        // return defaultUploadAction;
-        const exactMatches = updateResourcesOptions.filter(
-            x => x.fileName === file.name
-        );
-        switch (exactMatches.length) {
-            case 1:
-                return exactMatches[0];
-            default:
-                return extraResource;
+
+        function strDiff(str1, str2){
+            // Get all chars that match at the start of both strings
+            var openingMatch = "";
+            for(let x = 0; x < str1.length; x++){
+                if(str1.charAt(x) == str2.charAt(x)) openingMatch += str1.charAt(x);
+                else break;
+            }
+
+            // Get all chars that match at the end of both strings
+            var str1Reversed = str1.split("").reverse().join("");
+            var str2Reversed = str2.split("").reverse().join("");
+            var endingMatch = "";
+            for(let x = 0; x < str1.length; x++){
+                if(str1Reversed.charAt(x) == str2Reversed.charAt(x)) endingMatch += str1Reversed.charAt(x);
+                else break;
```

Comment:
This is very confusing. It should be:
```
if (str1Reversed.charAt(x) == str2Reversed.charAt(x)){
    endingMatch += str1Reversed.charAt(x);
}else{
    break;
}
```

Reply:
Technically the entire block (declaring a variable, looping and if-statment) can be put into a single `reduce()` statement